### ISSUE TITLE
[1.12.2]Crash - Updating Screen Events

### ISSUE THREAD

**Issue by Vyraal1** [codetaylor/artisan-worktables#47]
**Crash Log:** https://hastebin.com/iyivazanej.sql
**Description:** I'm not sure if all of these are related but I added Thermal Foundation Hammers to <ore:tooLHammer> oredictionary (which also includes the blacksmith_hammers). When I went to shift+click a thermal foundation platinum hammer into the Worktable, it seemed to have crashed. I can post the script if you want.

**Comment by JWTHDYTWA** [comment]
Not only TE hammers, it's exactly the same thing as it was before with me.

Check issue #41 

Here's new crashlog:
https://gist.github.com/Warrock3280/d03ee41f4364150f6a315d53fec7c168

**Comment by codetaylor** [comment]
Thanks for the reports. I'm looking into it.

This new issue has nothing to do with issue #41.

**Comment by codetaylor** [comment]
I'll bet this issue has something to do with the latest CraftTweaker update from 4.1.1 to 4.1.2.

I've seen other pack makers have issues with the new CT update.

Try stepping down to CT 4.1.1 and see if the issue persists. I'm looking into it on my end.

**Comment by codetaylor** [comment]
I was right.

Here is the CT commit that changes the line that is now crashing: https://github.com/jaredlll08/CraftTweaker/commit/a6396b6ef32edbf23005fc5817c2e3d3e84b16c4#diff-4e4c4842206dc45ace9cc24cbb4d3c88

I'll see what I can do about it.

**Comment by codetaylor** [comment]
@Vyraal1 Please post your script (just the relevant bits if you can). I'm unable to reproduce this, yet.

**Comment by Vyraal1** [comment]
Here's the relevant parts of the script - https://hastebin.com/upatexugax.xml
Here's also the global variables for plates - https://hastebin.com/nanizudapo.py

**Comment by codetaylor** [comment]
Finally, there it is!

```
Worktable.createRecipeBuilder("carpenter")
    .setShaped([
        [<ore:plankWood>],
        [<ore:plankWood>],
        [<ore:plankWood>]])
    .setTool(<ore:carpenters_hammer>, 30)
    .addOutput(<minecraft:gravel>)
    .create();
```

```
Caused by: java.lang.NullPointerException
	at crafttweaker.api.minecraft.CraftTweakerMC.matches(CraftTweakerMC.java:525) ~[CraftTweakerMC.class:?]
	at crafttweaker.api.minecraft.CraftTweakerMC.matches(CraftTweakerMC.java:533) ~[CraftTweakerMC.class:?]
	at crafttweaker.mc1120.oredict.MCOreDictEntry.contains(MCOreDictEntry.java:110) ~[MCOreDictEntry.class:?]
	at crafttweaker.mc1120.oredict.MCOreDictEntry.matches(MCOreDictEntry.java:180) ~[MCOreDictEntry.class:?]
	at com.codetaylor.mc.athenaeum.integration.crafttweaker.mtlib.helpers.CTInputHelper$IngredientWrapper.apply(CTInputHelper.java:379) ~[CTInputHelper$IngredientWrapper.class:?]
[...]
```

I know where the problem is and how to fix it. Shouldn't be long now.

**Comment by codetaylor** [comment]
Fixed in Athenaeum 1.12.2-1.6.4

**Comment by codetaylor** [comment]
https://minecraft.curseforge.com/projects/athenaeum/files/2527729

If you continue to experience this issue after updating the library, please contact me.
